text/0012-CLI-commands.md

```diff
@@ -0,0 +1,1366 @@
+  qri connect
+
+  # then, in a separate window, request the info from peer b5
+  qri info b5/comics
+```
+
+### Options
+
+```
+  -f, --format string   set output format [json]
+  -h, --help            help for info
+```
+
+
+
+<a id='qri_list'></a>
+## qri list
+
+Show a list of datasets
+
+### Synopsis
+
+
+List shows lists of datasets, including names and current hashes. 
+
+The default list is the latest version of all datasets you have on your local 
+qri repository.
+
+When used in conjuction with `qri connect`, list can list a peer's dataset. You
+must have `qri connect` running in a separate terminal window.
+
+```
+qri list [flags]
+```
+
+### Examples
+
+```
+  # show all of your datasets:
+  qri list
+
+  # to view the list of your peer's dataset,
+  # in one terminal window:
+  qri connect
+
+  # in a separate terminal window, to show all of b5's datasets:
+  qri list b5
+```
+
+### Options
+
+```
+  -f, --format string   set output format [json]
+  -h, --help            help for list
+  -l, --limit int       limit results, default 25 (default 25)
+  -o, --offset int      offset results, default 0
+```
+
+
+
+<a id='qri_log'></a>
+## qri log
+
+show log of dataset history
+
+### Synopsis
+
+
+log prints a list of changes to a dataset over time. Each entry in the log is a 
+snapshot of a dataset taken at the moment it was saved that keeps exact details 
+about how that dataset looked at at that point in time. 
+
+We call these snapshots versions. Each version has an author (the peer that 
+created the version) and a message explaining what changed. Log prints these 
+details in order of occurrence, starting with the most recent known version, 
+working backwards in time.
+
+```
+qri log [flags]
+```
+
+### Examples
+
+```
+  show log for the dataset b5/precip:
+  $ qri log b5/precip
+```
+
+### Options
+
+```
+  -h, --help         help for log
+  -l, --limit int    limit results, default 25 (default 25)
+  -o, --offset int   offset results, default 0
+```
+
+
+
+<a id='qri_new'></a>
+## qri new
+
+Create a new dataset
+
+### Synopsis
+
+
+New creates a dataset from data you supply. Please note that all data added
+to Qri is made public on the distributed web when you run qri connect
+
+When adding data, you can supply metadata and dataset structure, but it’s not
+required. qri does what it can to infer the details you don’t provide.
+qri currently supports three data formats:
+- CSV  (Comma Separated Values)
+- JSON (Javascript Object Notation)
+- CBOR (Concise Binary Object Representation)
+
+Once you’ve added data, you can use the export command to pull the data out of
+Qri, change the data outside of Qri, and use the save command to record those
+changes to Qri.
+
+Datasets added to Qri using `new` are published to the registry by default.
+Use the `--no-publish` flag to not publish this dataset to the registry. For
+more information on the registry and why it's important check out the [registry
+section](#qri_registry).
+
```

Comment:
Again, just deciding this. But very open to changing my mind, as I can see pros and cons for each.

pros: fill out the registry, have more stuff for folks to search, builds the network
cons: annoy or upset users that didn't realize their stuff was searchable by others

Although: Qri is public. It almost seems disingenuous to make people feel like by not 'publishing' their data, it is somehow shielded from the public eye. 

I have so many thoughts about this.

Reply:
so much this. I think my current feelings on the subject are here: https://github.com/qri-io/qri/issues/513#issuecomment-412281518

> After playing with this, I don't think qri new should publish by default. I know initially we'd agreed on the opposite, but after spending time using qri I've come to two major reasons for this:
> A lot of the datasets I make / work on are "practice" or unfinished work I don't want to announce to the world (yet).
> I often work without an internet connection, which creates inconsistent behaviour in regard to publishing
> In the offline example, qri publish running secretly in the background will fail in this context, which undermines the one upside of "always publish", which is being able to say "I ran new, it should be online". For these reasons, I think we should avoid publishing by default, and instead focus on building up a whole set of tutorials & flows around publication that's similar to git push.

This RFC process has highlighted two things that I think need more thought & clarification:
1. Getting registries right is going to be _very hard_. I think they should be as thin as possible
2. Our story around "remote control" should intersect with this `qri publish` conversation. Eventually we'll need something that's the rough equivalent of `git push`. (For example, an office has a qri node that's always online that you can "push to" so data gets copied there), which feels a lot like `qri publish`.

These are all questions that I'd like to resolve before putting `qri publish` in as a default behavior.

Reply:
oooo, great!! thank you.